### PR TITLE
Random Stuff

### PR DESCRIPTION
**Pull request by DavidVorick**
I overhauled the siad package and added some testing.

### CODE REVIEW

**Comment by DavidVorick** [comment]
Fixes #459

File: siad/main_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	}
+
+	testDir := tester.TempDir("siad", "TestMain")
+
+	// Try running and closing an instance of siad.
+	os.Args = []string{
+		"siad",
+		"-n",
+		"-a",
+		"localhost:45150",
```
Comment (lukechampine) [inline siad/main_test.go]: just use the port.

**Comment by lukechampine** [comment]
I don't agree with making `startDaemonCmd` a cobra command. It makes it harder to test, since it no longer returns an error that can be checked. Also, the use of a channel strikes me as ugly (and to be extra nit-picky, you send down the channel before you actually start serving requests).

My suggestion would be to have a cobra command that calls `createDaemon` and then starts the API server separately.

**Comment by DavidVorick** [comment]
It would just be a passthrough function, nothing more.

Sending down the channel has to happed before the server starts listening, because Serve() is a blocking function. I guess I could do `go startChan <- struct{}{}`

**Comment by DavidVorick** [comment]
My workflow got messed up and 2 extra commits squeezed through.

Some changes to the transaction pool to prevent transactions from being added to every block, and then a brief comment change + makefile change.

**Comment by lukechampine** [comment]
no, putting it in a go func doesn't help anything.
here's what I suggested:
```go
func startDaemonCmd(*cobra.Command, []string) {
    d, err := createDaemon()
    if err != nil {
        fmt.Println(err)
        return
    }
    startChan <- struct{}{}
    err := d.Serve()
    if err != nil {
        fmt.Println(err)
    }
}
```

really, `createDaemon` should take arguments instead of using global variables. Then you could test it without needing to pass flags to `main()`.

File: siad/main_test.go (hunk 1 above)
Comment (DavidVorick) [inline siad/main_test.go]: ?

**Comment by DavidVorick** [comment]
=/

hmph.

main() is my least favorite type of code

**Comment by lukechampine** [comment]
mine too :/

**Comment by DavidVorick** [comment]
putting it in a gofunc helps because the gofunc isn't going to trigger until later, so the code will at least get one step farther?

I guess Serve() is a block box.

**Comment by lukechampine** [comment]
the "right" way would be:
```go
go Serve()
startChan <- struct{}{}
select {} // or better, wait on a channel signalled by Serve()
```

**Comment by DavidVorick** [comment]
that's worse than sending down the channel in the go func, because you're almost guaranteed to have the `startChan <- struct{}{}` code run before the `Serve()` code.

The best way would be to push something down the channel inside of `Serve()`, but that's too much effort. The way it's happening now is working well enough.

**Comment by DavidVorick** [comment]
sorry, can't pull Serve() out of the `createDaemon` function. It breaks the flow and doesn't make sense to me.

**Comment by lukechampine** [comment]
what does that mean

**Comment by DavidVorick** [comment]
uhh, you'll see. I did my best but it's still not great.

**Comment by DavidVorick** [comment]
I have found the quest to give this more functional purity greatly frustrating and damaging to the code flow. I don't think that cobra was designed to operate in a pure functional environment (obviously, as it relies on global variables) and neither was gcfg (which relies on nested structs).

Trying to make things work this way has resulted in a lot of ugly modifications to the code and I don't want to keep the changes.

**Pull request merged**